testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -581,34 +536,28 @@ public void testReferencedTablesWithColumnMask()
         assertThat(tables).hasSize(2);
 
         TableInfo table = tables.get(0);
-        assertThat(table.getCatalog()).isEqualTo("tpch");
-        assertThat(table.getSchema()).isEqualTo("tiny");
-        assertThat(table.getTable()).isEqualTo("orders");
-        assertThat(table.getAuthorization()).isEqualTo("user");
-        assertThat(table.isDirectlyReferenced()).isFalse();
-        assertThat(table.getFilters()).isEmpty();
-        assertThat(table.getColumns()).hasSize(1);
-
-        ColumnInfo column = table.getColumns().get(0);
-        assertThat(column.getColumn()).isEqualTo("orderkey");
-        assertThat(column.getMask()).isEmpty();
+        assertThat(table)
+                .hasCatalogSchemaTable("tpch", "tiny", "orders")
+                .hasAuthorization("user")
+                .isDirectlyReferenced(false)
+                .hasColumnsWithoutMasking("orderkey")
+                .hasFilters();
 
         table = tables.get(1);
-        assertThat(table.getCatalog()).isEqualTo("mock");
-        assertThat(table.getSchema()).isEqualTo("default");
-        assertThat(table.getTable()).isEqualTo("test_table_with_column_mask");
-        assertThat(table.getAuthorization()).isEqualTo("user");
-        assertThat(table.isDirectlyReferenced()).isTrue();
-        assertThat(table.getFilters()).isEmpty();
-        assertThat(table.getColumns()).hasSize(2);
-
-        column = table.getColumns().get(0);
-        assertThat(column.getColumn()).isEqualTo("test_varchar");
-        assertThat(column.getMask()).isPresent();
-
-        column = table.getColumns().get(1);
-        assertThat(column.getColumn()).isEqualTo("test_bigint");
-        assertThat(column.getMask()).isEmpty();
+        assertThat(table)
+                .hasCatalogSchemaTable("mock", "default", "test_table_with_column_mask")
+                .hasAuthorization("user")
+                .isDirectlyReferenced(true)
+                .hasColumnsSatisfying(
+                        ci -> {
+                            assertThat(ci.getColumn()).isEqualTo("test_varchar");
+                            assertThat(ci.getMask()).hasValueSatisfying(m -> assertThat(m).isEqualToIgnoringWhitespace("(SELECT CAST(max(orderkey) AS varchar(15)) FROM orders)"));
+                        },
+                        ci -> {
+                            assertThat(ci.getColumn()).isEqualTo("test_bigint");
+                            assertThat(ci.getMask()).isEmpty();
+                        })
```

Comment:
Is it possible to extract them as `hasColumnMask(String columnName, String maskInformation)` ?

Reply:
Not exactly, but I refactored a bit so that you can avoid having to write assertions directly. Split it out into `hasColumnNames` and `hasColumnMasks`